
From: William Lee Irwin III <wli@holomorphy.com>

fs/reiser4/plugin/pseudo/pseudo.c: In function `get_rwx':
fs/reiser4/plugin/pseudo/pseudo.c:603: warning: comparison is always false due to limited range of data type

This one is a real bug.

Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/fs/reiser4/plugin/pseudo/pseudo.c |    2 +-
 1 files changed, 1 insertion(+), 1 deletion(-)

diff -puN fs/reiser4/plugin/pseudo/pseudo.c~reiser4-mode-fix fs/reiser4/plugin/pseudo/pseudo.c
--- 25/fs/reiser4/plugin/pseudo/pseudo.c~reiser4-mode-fix	Tue Aug 24 15:52:03 2004
+++ 25-akpm/fs/reiser4/plugin/pseudo/pseudo.c	Tue Aug 24 15:52:03 2004
@@ -600,7 +600,7 @@ static int get_rwx(struct file *file, co
 			struct iattr newattrs;
 
 			down(&host->i_sem);
-			if (rwx == (mode_t) -1)
+			if (rwx == (umode_t)~0)
 				rwx = host->i_mode;
 			newattrs.ia_mode =
 				(rwx & S_IALLUGO) | (host->i_mode & ~S_IALLUGO);
_
